R/France.R

```diff
@@ -0,0 +1,126 @@
+#' France Class containing country specific attributes and methods
+#'
+#' @description Country specific information for downloading, cleaning
+#' and processing covid-19 region data for France.
+#'
+#' @details Inherits from `DataClass`
+#'
+# nolint start
+#' @source https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675
+#' @source https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c
+#' @source https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01
+# nolint end
+#' @export
+#' @examples
+#' \dontrun{
+#' region <- France$new(verbose = TRUE, steps = TRUE, get = TRUE)
+#' region$return()
+#' }
+#'
+France <- R6::R6Class("France",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field country name of country to fetch data for
+    country = "France",
+    #' @field supported_levels A list of supported levels.
+    supported_levels = list("1", "2"),
+    #' @field supported_region_names A list of region names in order of level.
+    supported_region_names = list("1" = "region", "2" = "department"),
+    #' @field supported_region_codes A list of region codes in order of level.
+    supported_region_codes = list("1" = "iso_3166_2", "2" = "ons_region_code"),
+    #' @field common_data_urls List of named links to raw data. This is empty
+    #' as for France data is divided into cases by level
+    common_data_urls = list(),
+    # nolint start
+    #' @field level_data_urls List of named links to raw data that are level
+    #' specific.
+    level_data_urls = list(
+      "1" = list("cases" = "https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01"),
+      "2" = list(
+        "cases" = "https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675",
+        "hosp" = "https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c"
+      )
+    ),
+    # nolint end
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "tested_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble
+    #' @importFrom dplyr select
+    set_region_codes = function() {
+      if (self$level == "1") {
```

Comment:
I've been working under the assumption we would just assign all the lookups all of the time so they are generally available across levels (for example adding level 1 codes to level 2 data might be nice when nested)